### PR TITLE
Implement policies based on chunk creation time

### PR DESCRIPTION
**Pull request by pdipesh02**
- The idea with these changes is to support INTERVAL boundaries for
  INTEGER type columns in policy APIs. The boundary calculation will
  be done using chunk creation time metadata.
- Added a new index on hypertable_id and creation_time on timescaledb
  catalog table CHUNK.
- Added access methods to perform index scan on CHUNK table using the
  index created on hypertable_id and creation_time.
- Added ChunkVec to accumulated the chunks satisfying the range
  criteria during index scan.
- Updated show_chunks, drop_chunks and retention policy APIs to get the
  affected chunks using chunk creation time metadata based on the
  INTERVAL boundary specified for the INTEGER columns.

### CODE REVIEW

File: src/chunk.c [hunk 1]
```diff
@@ -4871,3 +4875,9 @@ ts_chunk_get_osm_chunk_id(int hypertable_id)
 
 	return chunk_id;
 }
+
+TimestampTz
+ts_chunk_get_creation_time(Oid chunk_oid)
+{
+	return GetCurrentTimestamp();
+}
```
Comment (fabriziomello) [inline src/chunk.c:4883]: Leftover?

File: sql/updates/latest-dev.sql [hunk 2]
```diff
@@ -0,0 +1,92 @@
+-- add a new column creation_time to chunk catalog table
+ALTER TABLE _timescaledb_catalog.chunk ADD COLUMN creation_time TIMESTAMPTZ;
+UPDATE _timescaledb_catalog.chunk SET creation_time = now();
```
Comment (fabriziomello) [inline sql/updates/latest-dev.sql:3]: I guess we can do a better assumption here using for example `pg_stat_file` function, something like:

```suggestion
UPDATE
    _timescaledb_catalog.chunk c
SET
    creation_time = (pg_catalog.pg_stat_file(pg_catalog.pg_relation_filepath(r.oid))).change
FROM
    pg_class r, pg_namespace n
WHERE
    r.relnamespace = n.oid
    AND r.relname = c.table_name
    AND n.nspname = c.schema_name
    AND r.relkind = 'r'
    AND c.dropped IS FALSE;
UPDATE _timescaledb_catalog.chunk SET creation_time = now() WHERE creation_time IS NULL;
```
The second UPDATE is just to make sure we'll don't left any record without creation time information. One improvement is try to use pg_stat_file.creation field when the update run on Windows platform. https://www.postgresql.org/docs/15/functions-admin.html

File: src/chunk.c (hunk 1 above)
Comment (pdipesh02) [reply src/chunk.c:4883]: Yes, this needs to integrated with follow up PR on policy implementation.

**Comment by nikkhils** [comment]
This is being further tracked via https://github.com/timescale/timescaledb/pull/6177

**Pull request closed without merging**